packages/core/src/schema/types.ts

```diff
@@ -44,24 +47,64 @@ export type RunErrorInsert = InferInsertModel<typeof runErrors>
 export type Evaluation = InferSelectModel<typeof evaluations>
 export type ConnectedEvaluation = InferSelectModel<typeof connectedEvaluations>
 export type EvaluationResult = InferSelectModel<typeof evaluationResults>
-export type EvaluationTemplate = InferSelectModel<typeof evaluationTemplates>
+export type EvaluationTemplate = InferSelectModel<
+  typeof evaluationLegacyTemplates
+>
 export type MagicLinkToken = InferSelectModel<typeof magicLinkTokens>
 export type ClaimedReward = InferSelectModel<typeof claimedRewards>
 export type EvaluationTemplateCategory = InferSelectModel<
   typeof evaluationTemplateCategories
 >
-export type LlmAsJudgeEvaluationMetadata = InferSelectModel<
-  typeof llmAsJudgeEvaluationMetadatas
+export type EvaluationMetadataLlmAsJudgeLegacy = InferSelectModel<
+  typeof evaluationMetadataLlmAsJudgeLegacy
+>
+export type EvaluationMetadataLlmAsJudgeBoolean = InferSelectModel<
+  typeof evaluationMetadataLlmAsJudgeBoolean
+>
+export type EvaluationMetadataLlmAsJudgeNumerical = InferSelectModel<
+  typeof evaluationMetadataLlmAsJudgeNumerical
+>
+export type EvaluationMetadataLlmAsJudgeCustom = InferSelectModel<
+  typeof evaluationMetadataLlmAsJudgeCustom
 >
-export type Subscription = InferSelectModel<typeof subscriptions>
 
-export type EvaluationDto = Evaluation & {
+export interface EvaluationDtoLlmAsJudgeLegacy extends Evaluation {
+  metadataType: EvaluationMetadataType.LlmAsJudgeLegacy
+  metadata: Omit<
+    EvaluationMetadataLlmAsJudgeLegacy,
+    'metadataType' | 'createdAt' | 'updatedAt'
+  >
+}
+export interface EvaluationDtoLlmAsJudgeBoolean extends Evaluation {
+  metadataType: EvaluationMetadataType.LlmAsJudgeBoolean
+  metadata: Omit<
+    EvaluationMetadataLlmAsJudgeBoolean,
+    'metadataType' | 'createdAt' | 'updatedAt'
+  >
+}
+export interface EvaluationDtoLlmAsJudgeNumerical extends Evaluation {
+  metadataType: EvaluationMetadataType.LlmAsJudgeNumerical
+  metadata: Omit<
+    EvaluationMetadataLlmAsJudgeNumerical,
+    'metadataType' | 'createdAt' | 'updatedAt'
+  >
+}
+export interface EvaluationDtoLlmAsJudgeCustom extends Evaluation {
+  metadataType: EvaluationMetadataType.LlmAsJudgeCustom
   metadata: Omit<
-    LlmAsJudgeEvaluationMetadata,
+    EvaluationMetadataLlmAsJudgeCustom,
     'metadataType' | 'createdAt' | 'updatedAt'
   >
 }
 
+export type EvaluationDto =
+  | EvaluationDtoLlmAsJudgeLegacy
+  | EvaluationDtoLlmAsJudgeBoolean
+  | EvaluationDtoLlmAsJudgeNumerical
+  | EvaluationDtoLlmAsJudgeCustom
```

Comment:
This is the new `EvaluationDto` type. It contains everything from `Evaluation`, with an additional `metadata` object which contains the correspondent `EvaluationMetadata` fields depending on the `evaluations.metadataType`

Reply:
(moved to future PR)